### ISSUE TITLE
Language tag appended in literal part of a lang string.

### ISSUE THREAD

**Issue by fanavarro** [ontop/ontop#242]
Hi all,

I'm seeing a strange behavior when I query about literals with a language tag established. Ontop returns an OWLLiteralImplPlain with the property "lang" filled correctly, but this language tag is appended to the literal value too. For example, the following query about the rdfs:comment of an ontology class

```
SELECT ?instancia ?comment  WHERE {
	VALUES ?instancia { <http://www.basecia.es/ontologia#CuentaContableActivos> } .
	?instancia <http://www.w3.org/2000/01/rdf-schema#comment> ?comment .
}
```
returns a OWLLiteralImplPlain with the following properties:

- lang = "es"

- literal = "Cuenta bancaria interna de activos.es"

As you can see, the language tag "es" is appended to the literal part.

I'm using the code from the main branch of git repository.
I'm going to do some debug. I'll write if I found something.

The files used (ontology, mappings and properties) are attached.

Cheers,
Fran.

[obda.zip](https://github.com/ontop/ontop/files/1673140/obda.zip)

**Comment by fanavarro** [comment]
I've found that a lang string is represented as a function. In my case, the following one:
`http://www.w3.org/1999/02/22-rdf-syntax-ns#langString("Cuenta bancaria interna de activos.","es")`
This function has 2 terms: the literal part ("Cuenta bancaria interna de activos.") and the language tag ("es").

The method "getSQLString(Term term, AliasIndex index, boolean useBrackets)" from OneShotSQLGeneratorEngine detects the type of the function as data type function. After that, getSQLStringForTemplateFunction is called. In this function, the SQL of the terms are concatenated so, finally, we have the following SQL line:

`('Cuenta bancaria interna de activos.' || REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(CAST('es' AS VARCHAR(4000)), ' ', '%20'), '!', '%21'), '''', '%22'), '#', '%23'), '$', '%24'), '&', '%26'), '(', '%28'), ')', '%29'), '*', '%2A'), '+', '%2B'), ',', '%2C'), '/', '%2F'), ':', '%3A'), ';', '%3B'), '=', '%3D'), '?', '%3F'), '@', '%40'), '[', '%5B'), ']', '%5D'))`

Then, the final query is: 
```
SELECT 
   1 AS "instanciaQuestType", NULL AS "instanciaLang", 'http://www.basecia.es/ontologia#CuentaContableActivos' AS "instancia", 
   -3 AS "commentQuestType", 'es' AS "commentLang", ('Cuenta bancaria interna de activos.' || REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(CAST('es' AS VARCHAR(4000)), ' ', '%20'), '!', '%21'), '''', '%22'), '#', '%23'), '$', '%24'), '&', '%26'), '(', '%28'), ')', '%29'), '*', '%2A'), '+', '%2B'), ',', '%2C'), '/', '%2F'), ':', '%3A'), ';', '%3B'), '=', '%3D'), '?', '%3F'), '@', '%40'), '[', '%5B'), ']', '%5D')) AS "comment"
FROM 
    (SELECT 1 from dual) tdummy
```
As you can see, the language tag is appended in the column "comment".

Could a check for langString functions be necessary  to avoid this concatenation in "getSQLString" function of OneShotSQLGeneratorEngine?

Greetings,
Fran.

**Comment by fanavarro** [comment]
Hi, I've developed a quick fix for this (#243 ), treating the langString as a function with only one term to avoid the term concatenation. I've tested my code and it works well, but I don't know if there is any reason to treat both terms of langString function.

Greetings,
Fran.

**Comment by juliencorman** [comment]
Fixed by bceaba3b (pull request), followed by dfe1585a
